tcms/testcases/forms.py

```diff
@@ -184,7 +184,7 @@ class BaseCaseSearchForm(forms.Form):
     tag__name__in = forms.CharField(required=False)
     category = forms.ModelChoiceField(
         label="Category",
-        queryset=Category.objects.none(),
+        queryset=Category.objects.all(),
```

Comment:
Categories must be filtered by TestPlan.product.

If you had tried to inspect the form visually you would have seen this.

Reply:
In fact the initial values are correct and the filtering is applied in `BaseCaseSearchForm.populate()`.

Instead of changing the defaults each view that uses these forms must call `form.populate()` with the product ID to update the values for these select fields.